tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml

```diff
@@ -41,7 +41,29 @@ steps:
       # available at runtime. Also, using ! as separator instead of the usual / because forward slash is something we
       # need to replace.
       TEST_PACKAGE_NAME=$(echo "${{ parameters.testPackageName }}" | sed -r 's!@!!g' | sed -r 's!/!-!g')
-      echo "##vso[task.setvariable variable=itpbip_sanitizedPackageName]${TEST_PACKAGE_NAME}-*.*.*-*.tgz"
+
+      # There's some thought behind the value for the itpbip_sanitizedPackageName to make sure that two packages with
+      # partially overlapping names (e.g. @fluid-experimental/tree and @fluid-experimental/tree-react-api) can be
+      # differentiated, so the pipeline knows which of the two tgz files it is supposed to unpack, while also avoiding
+      # these issues we've run into in the past:
+      # - A pattern like `${TEST_PACKAGE_NAME}-?.?.?-*.tgz` stops working for two-digit major/minor/patch version numbers.
+      #   It also doesn't work when the pipeline runs for a release/* branch because in that case the tgz filename looks
+      #   like `<package-name>-<major>.<minor>.<patch>.tgz` instead of `<package-name>-<major>.<minor>.<patch>-<build-id>.tgz`,
+      #   so the final dash in the pattern causes it to not match.
+      # - A pattern like `${TEST_PACKAGE_NAME}-*.*.*-*.tgz` won't differentiate packages in all cases because extra
+      #   parts of the package name are separated with dashes (e.g.fluidexperimental-tree and fluidexperimental-tree-react-api),
+      #   which are also used to separate the package name from the major version number.
+      # The logic behind the current pattern is that we want to find these components:
+      # - `${TEST_PACKAGE_NAME}` - Matching package name.
+      # - `-` A dash.
+      # - `+([0-9]).+([0-9]).+([0-9])` - The version (i.e. three sets of one or more digits separated by dots).
+      #   This component *must not* match another bit of package name. As long as we don't put numbers in package names,
+      #   the current pattern should keep working.
+      # - `*` - Potentially more characters (to handle the case where "-<build-id>" is appended to the version number).
+      # - `.tgz` - The extension.
+      # NOTE: this is not standard regex syntax, but ADO globbing.
+      # See https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/file-matching-patterns?view=azure-devops#pattern-syntax for details.
```

Comment:
I thought there was a use case where the glob was used with `ls` command. At least I have a comment relating to `ls` in Teams discussion thread.

Reply:
Thanks for the callout, I had forgotten about that and I think you're right. I'll confirm and update this.